secretstores/tencent/ssm/ssm.go

```diff
@@ -0,0 +1,189 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common"
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common/profile"
+	ssm "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/ssm/v20190923"
+
+	"github.com/dapr/components-contrib/secretstores"
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	SecretID         = "SecretID"
+	SecretKey        = "SecretKey"
+	Token            = "Token"
+	Region           = "Region"
+	VersionID        = "VersionID"
+	RequestID        = "RequestID"
+	ValueType        = "SecretValueType"
+	pageLimit uint64 = 100
+)
+
+type SecretValueType uint16
+
+var (
+	// TEXT_SECRET_VALUE text secret.
+	TEXT_SECRET_VALUE SecretValueType = 10
+	// BINARY_SECRET_VALUE binary secret
+	BINARY_SECRET_VALUE SecretValueType = 20
+)
+
+type ssmClient interface {
+	ListSecretsWithContext(ctx context.Context, request *ssm.ListSecretsRequest) (*ssm.ListSecretsResponse, error)
+	GetSecretValueWithContext(ctx context.Context, request *ssm.GetSecretValueRequest) (*ssm.GetSecretValueResponse, error)
+}
+
+type ssmSecretStore struct {
+	client ssmClient
+	logger logger.Logger
+}
+
+var _ secretstores.SecretStore = (*ssmSecretStore)(nil)
+
+// NewSSM returns a new TencentCloud ssm secret store.
+func NewSSM(logger logger.Logger) secretstores.SecretStore {
+	return &ssmSecretStore{
+		logger: logger,
+	}
+}
+
+// Init creates a TencentCloud ssm client.
+func (s *ssmSecretStore) Init(metadata secretstores.Metadata) error {
+	var (
+		err    error
+		region string
+	)
+
+	secretID := metadata.Properties[SecretID]
+	secretKey := metadata.Properties[SecretKey]
+	token := metadata.Properties[Token]
+	region = metadata.Properties[Region]
+	if secretID == "" || secretKey == "" {
+		return errors.New("secret params are empty")
+	}
+
+	credential := common.NewTokenCredential(secretID, secretKey, token)
+	s.client, err = ssm.NewClient(credential, region, profile.NewClientProfile())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetSecret retrieves a secret using a key and returns a map of decrypted string/string values.
+func (s *ssmSecretStore) GetSecret(ctx context.Context, req secretstores.GetSecretRequest) (secretstores.GetSecretResponse, error) {
+	response := secretstores.GetSecretResponse{
+		Data: map[string]string{},
+	}
+	if req.Name == "" {
+		return response, errors.New("secret name is empty")
+	}
+
+	versionID := req.Metadata[VersionID]
+	ssmReq := ssm.NewGetSecretValueRequest()
+	ssmReq.SecretName = &req.Name
+	ssmReq.VersionId = &versionID
+	ssResp, err := s.client.GetSecretValueWithContext(ctx, ssmReq)
+	if err != nil {
+		return response, err
+	}
+	var val string
+	ssResponse := ssResp.Response
+	if ssResponse.SecretBinary != nil {
+		response.Data[ValueType] = fmt.Sprintf("%d", BINARY_SECRET_VALUE)
```

Comment:
Nit: `strconv.Itoa(BINARY_SECRET_VALUE)` would be much faster

Reply:
Ok. Thanks.